qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -166,11 +244,33 @@ def settings(self) -> Dict[str, Any]:
         """
         raise NotImplementedError("The settings method is not implemented per default.")
 
-    @abstractmethod
+    # TODO make abstract after the deprecation period
+    # @abstractmethod
```

Comment:
Because users might have their own optimizers that derive from `Optimizer`. If we make this abstract, their code will complain that `minimize` doesn't exist 🙂

Reply:
That is true. But then VQE has been changed to call `minimize` has it not - so it would fail if it were passed an optimizer in the old form no? I guess it can try and call `mimimize` (which exists in the base class as non abstract) and catch the NotImplementedError and then call `optimize` warning that the optimizer needs to be updated.

Reply:
Good point, we should add that safeguard in VQE 👍🏻